### PR TITLE
[14.0][FIX] purchase_operating_unit: default picking_type without OU

### PR DESCRIPTION
**Pull request by HekkiMelody**
Fixes an issue when creating a PO, if the user doesn't have any operating units assigned.

### CODE REVIEW

File: purchase_operating_unit/models/purchase_order.py [hunk 1]
```diff
@@ -14,14 +14,15 @@ def _default_picking_type(self):
...
-            res = types[:1].id
+        if operating_unit:
+            types = type_obj.search(
+                [
+                    ("code", "=", "incoming"),
+                    ("warehouse_id.operating_unit_id", "=", operating_unit.id),
+                ]
+            )
+            if types:
+                res = types[0].id
```
Comment (AaronHForgeFlow) [inline purchase_operating_unit/models/purchase_order.py]: Before it was `res = types[:1].id` why this change?

Comment (HekkiMelody) [reply purchase_operating_unit/models/purchase_order.py]: Seemed cleaner to me, but I can revert it.

`[:1]` is useful to make sure there are no errors on an empty recordset. But we already have `if types`, and also it would error out when trying to access `.id` anyway.

Either way, that's not the point of the issue, so I'll remove it.

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

@AaronHForgeFlow good for you?

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM. Thanks for the explanation

**Comment by AaronHForgeFlow** [comment]
/ocabot merge patch

If you do the FW to 15.0 it would merge it right away

**Comment by HekkiMelody** [comment]
Thank you @AaronHForgeFlow !

**Comment by HekkiMelody** [comment]
The method I'm fixing isn't even present in v15, I'm afraid it'll require to FW multiple commits to align the versions...

**Comment by francesco-ooops** [comment]
@AaronHForgeFlow can you make an analysis of commits in v14 missing in v15?

**Comment by AaronHForgeFlow** [comment]
hmmm, theoretically every PR merged is forward ported to new versions. I will check when I have some time
